mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp

```diff
@@ -617,6 +625,68 @@ bool CNEOBaseCombatWeapon::CanBePickedUpByClass(int classId)
 	return true;
 }
 
+#ifdef CLIENT_DLL
+void CNEOBaseCombatWeapon::ProcessMuzzleFlashEvent()
+{
+	if (GetPlayerOwner() == NULL)
+		return; // If using a view model in first person, muzzle flashes are not processed until the player drops their weapon. In that case, do not play a muzzle flash effect. Need to change how this is calculated if we want to allow dropped weapons to cook off for example
+
+	if ((GetNeoWepBits() & NEO_WEP_SUPPRESSED))
+		return;
+
+	int iAttachment = -1;
+	if (GetBaseAnimating()) {
+		// Find the attachment point index
+		iAttachment = GetBaseAnimating()->LookupAttachment("muzzle_flash");
+		if (iAttachment <= 0)
+		{
+			Warning("Model '%s' doesn't have attachment '%s'\n", GetPrintName(), "muzzle_flash");
+			return;
+		}
+	}
+
+	// Muzzle flash light
+	Vector vAttachment;
+	QAngle dummyAngles;
+	GetAttachment(iAttachment, vAttachment, dummyAngles);
+
+	// Make an elight
+	dlight_t* el = effects->CL_AllocElight(LIGHT_INDEX_MUZZLEFLASH + index);
+	el->origin = vAttachment;
+	el->radius = random->RandomInt(32, 64);
+	el->decay = el->radius / 0.05f;
+	el->die = gpGlobals->curtime + 0.05f;
+	el->color.r = 255;
+	el->color.g = 192;
+	el->color.b = 64;
+	el->color.exponent = 5;
+
+	// Muzzle flash particle
+	DispatchMuzzleParticleEffect(iAttachment);
+}
+
+void CNEOBaseCombatWeapon::DispatchMuzzleParticleEffect(int iAttachment) {
```

Comment:
![image](https://github.com/NeotokyoRebuild/neo/assets/44210793/4ae2801b-7226-452a-9163-892e9cfd00df)
Good find, was supposed to pass this to DispatchEffect via data. Since most 3rd person weapon models only have one attachment point (the muzzle_flash), it still worked before